### PR TITLE
Adding LN.Pizza to vendor list

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR contains the addition of ln.pizza to vendor list.
Currently, it's not possible to show an interactable AlertDialog on top of Webview plugin and ln.pizza requires the user to come back to the payment screen after the payment is complete. We overcome this issue by screenshotting the webview's visible content and showing it as a widget and hiding the webview till payment is complete. This creates the illusion that the payment request dialog is on top of webview.

Incomplete:
 - Test for successful payments, ~Bitrefill needs to rebalance the channel we have with them.~ Akiba opened a new channel with Bitrefill.
   - ~I predict issues around collapse animation because of the usage of firstPaymentItemKey.~ There were no issues with collapse animation.
 - Keep showing screenshot during the screenshot to Webview transition if possible.
 - VendorRow needs a more generic solution.

### CODE REVIEW

File: lib/bloc/marketplace/marketplace_bloc.dart [hunk 1]
```diff
@@ -20,8 +20,9 @@ class MarketplaceBloc {
     _vendorData.forEach((vendorOptions) {
       String _url = config.get(vendorOptions, 'url');
       String _logo = config.get(vendorOptions, 'logo');
-      VendorModel _vendorModel =
-      VendorModel(_url, vendorOptions, logo: _logo);
+      String _onlyShowLogoStr = config.get(vendorOptions, 'onlyShowLogo') ?? 'false';
```
Comment (roeierez) [inline lib/bloc/marketplace/marketplace_bloc.dart]:
```suggestion
      String _onlyShowLogo = config.get(vendorOptions, 'onlyShowLogo') == "true"
```

**Comment by roeierez** [comment]
@erdemyerebasmaz where is the implementation of webln script injection?

**Comment by erdemyerebasmaz** [comment]
> @erdemyerebasmaz where is the implementation of webln script injection?

I'll open a separate PR on plugin repo

**Comment by erdemyerebasmaz** [comment]
> @erdemyerebasmaz where is the implementation of webln script injection?

Here is the webln script injection: https://github.com/breez/breezmobile/pull/108/commits/e78eecc10ddeb33575d66b36ffc8f4b67202c26f

File: lib/bloc/marketplace/marketplace_bloc.dart (hunk 1 above)
Comment (erdemyerebasmaz) [reply lib/bloc/marketplace/marketplace_bloc.dart]: @roeierez Resolved at https://github.com/breez/breezmobile/pull/108/commits/90ec3bacc59443b2303bbce982c78df12dc7e392

File: lib/routes/user/marketplace/vendor_webview.dart [hunk 2]
```diff
@@ -64,10 +64,80 @@ class VendorWebViewPageState extends State<VendorWebViewPage> {
...
+        "       window.postMessage(JSON.stringify({ uri: paymentRequest }));\n" +
+        "       processedInvoices.push(paymentRequest);\n" +
+        "       return new Promise(function (resolve, reject) { });\n" +
+        "   },\n" +
+        "};\n" +
+        "setTimeout(function () { WebLN.requestProvider().catch(e => console.log(e));},450);";
+    _widgetWebview.evalJavascript(initWebLN).whenComplete(() => _findBolt11Links());
+  }
+
+  _findBolt11Links() {
```
Comment (roeierez) [inline lib/routes/user/marketplace/vendor_webview.dart]: This is not in webln standard. I don't think we should add this dynamic search for links here. Let's stick to webln interface.

File: lib/routes/user/marketplace/vendor_webview.dart [hunk 3]
```diff
@@ -64,10 +64,80 @@ class VendorWebViewPageState extends State<VendorWebViewPage> {
...
+    String includeWebLN = "var webLNLib = document.createElement('script');\n" +
+        "webLNLib.src = 'https://unpkg.com/webln@0.2.0/dist/webln.min.js';\n" +
+        "webLNLib.integrity = 'sha384-mTReBqbhPO7ljQeIoFaD1NYS2KiYMwFJhUNpdwLj+VIuhhjvHQlZ1XpwzAvd93nQ';\n" +
+        "webLNLib.crossOrigin = 'anonymous';\n" +
+        "document.body.appendChild(webLNLib);";
+    _widgetWebview.evalJavascript(includeWebLN).whenComplete(() => _initializeWebLN());
+  }
+
+  _initializeWebLN() {
+    String initWebLN = "   var processedInvoices = [];\n" +
```
Comment (roeierez) [inline lib/routes/user/marketplace/vendor_webview.dart]: I think it is good idea to place the script in its own resource file. Makes it better to maintain and read.

File: lib/routes/user/marketplace/vendor_webview.dart [hunk 4]
```diff
@@ -64,10 +64,80 @@ class VendorWebViewPageState extends State<VendorWebViewPage> {
...
+        "       webln = {\n" +
+        "   enable: function () {\n" +
+        "       window.postMessage(JSON.stringify({ enable: true }));\n" +
+        "       return new Promise(function (resolve, reject) {resolve(true);});\n" +
+        "   },\n" +
+        "   sendPayment: function (paymentRequest) {\n" +
+        "       window.postMessage(JSON.stringify({ uri: paymentRequest }));\n" +
+        "       processedInvoices.push(paymentRequest);\n" +
+        "       return new Promise(function (resolve, reject) { });\n" +
```
Comment (roeierez) [inline lib/routes/user/marketplace/vendor_webview.dart]: The standard states that we should either resolve or reject the promise according to if the user has payed successfully or failed/cancelled.
To achieve that I suggest to do the following:
1. Maintain a global mapping of requestID -> pendingPromise (on window object) and register a new request in each sendPayment.
2. In the postMessage of sendPayment add property requestID to the data sent with unique (can be auto-increment) number. 
3. From the webview on getting the sendPayment post message keep the requestID as pending request (We can assume one payment at a time).
4. On  fulfilled payments do something similar to: `_widgetWebview.evalJavascript("window.resolveRequest(requestID)")`
5. On  canceled and failed payments do something similar to: `_widgetWebview.evalJavascript("window.rejectRequest(requestID)")`
6. In javascript map the requestID to the request promise using the global mapping, call either reject or resolve and remove the mapping for this request.

**Pull request merged**